std/range/package.d

```diff
@@ -9213,3 +9221,302 @@ if (is(typeof(fun) == void) || isSomeFunction!fun)
 
     auto r = [1, 2, 3, 4].tee!func1.tee!func2;
 }
+
+/**
+Extend the length of the input range $(D r) by padding out the start of the
+range with the element $(D e). The element $(D e) must be of a common type with
+the element type of the range $(D r) as defined by $(D std.traits.CommonType).
+If $(D r) is an input range, this function is $(BIGOH 1). But if it is a forward
+range, it's $(BIGOH r.length).
+
+Params:
+    r = an input range with a length, or a string, or a forward range
+    e = element to pad the range with
+    n = the length to pad to
+
+Return:
+    a range containing the elements of the original range with the extra padding
+*/
+auto padLeft(R, E)(R r, E e, size_t n) if (
+    ((isInputRange!R && hasLength!R) || isNarrowString!R || isForwardRange!R) &&
+    !is(CommonType!(ElementType!R, E) == void))
+{
+    static if (hasLength!R || isNarrowString!R)
+    {
+        return e.repeat(n > r.length ? n - r.length : 0).chain(r);
+    }
+    else
+    {
+        size_t data_length = r.save.walkLength;
```

Comment:
No need to walk the entire range, just up to `n`. Also you may want to reflect that in the documentation as well.